src/map/status.c

```diff
@@ -1041,10 +1044,9 @@ static int status_check_skilluse(struct block_list *src, struct block_list *targ
 				return 0;
 			if (tsc != NULL) {
 				if (tsc->option&hide_flag
-				 && !is_boss
-				 && ((tsd->special_state.perfect_hiding || !is_detect)
-				  || (tsc->data[SC_CLOAKINGEXCEED] != NULL && is_detect)
-				))
+				&& !is_boss
+				&& ((tsd->special_state.perfect_hiding || !is_detect) || ((tsc->data[SC_CLOAKINGEXCEED] != NULL || tsc->data[SC_NEWMOON] != NULL) && is_detect))
```

Comment:
this condition can be simplified into this
``&& (tsd->special_state.perfect_hiding || !is_detect || tsc->data[SC_CLOAKINGEXCEED] != NULL || tsc->data[SC_NEWMOON] != NULL)``